storage/remote/azuread/azuread_config.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azuread
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+)
+
+// AzureAdConfig is the configuration for getting the accessToken
+// for remote write requests to Azure Monitoring Workspace
+type AzureAdConfig struct {
+	// AzureClientId is the clientId of the managed identity that is being used to authenticate.
+	AzureClientId string `yaml:"azure_client_id,omitempty"`
```

Comment:
nit:
```suggestion
	ClientID string `yaml:"client_id,omitempty"`
```

Reply:
done

Reply:
@rakshith210 , will we consider supporting users setting the endpoint url for retrieving aad tokens from the client_id?
We have use cases where instead of using the imds url, we have our own endpoint url for fetching the aad tokens, it'll be really helpful if you could add the endpoint_url, or hostname to the AzureAdConfig.

Reply:
Yes I do plan to add AAD based auth in the next iteration. In this PR, we wanted to have changes limited to just one auth (managed identity in this case).